app.js

```diff
@@ -9,8 +9,25 @@ app.get("/api", getApi);
 
 app.get("/api/topics", getTopics);
 
+app.get("/api/articles/:article_id", getArticleById);
+
 app.all("*", (req, res) => {
   res.status(404).send({ msg: "Not Found" });
 });
 
+app.use((err, req, res, next) => {
+  // postgres errors
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Bad request" });
+  } else {
+    next(err);
+  }
+});
+app.use((err, req, res, next) => {
+  // custom errors
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+  }
+});
+
```

Comment:
nice work with these, you can extract them out into another file just like your controllers!